src/Symfony/Component/HttpKernel/Controller/Layout/GenericControllerLayout.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\Layout;
+
+use Symfony\Component\HttpKernel\Controller\ActionReference;
+use Symfony\Component\HttpKernel\Controller\ControllerLayoutInterface;
+use Symfony\Component\HttpKernel\Exception\ControllerLayoutException;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+/**
+ * Class for generic *Bundle/Controller/*Controller::*Action layout.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Pavel Batanov <pavel@batanov.me>
+ */
+final class GenericControllerLayout implements ControllerLayoutInterface
+{
+    /**
+     * @var KernelInterface
+     */
+    private $kernel;
+
+    /**
+     * GenericControllerLayout constructor.
+     *
+     * @param KernelInterface $kernel
+     */
+    public function __construct(KernelInterface $kernel)
+    {
+        $this->kernel = $kernel;
+    }
+
+    public function parse(string $controller): ActionReference
+    {
+        if (0 === preg_match('#^(.*?\\\\Controller\\\\(.+)Controller)::(.+)Action$#', $controller, $match)) {
+            throw new \InvalidArgumentException(
+                sprintf('The "%s" controller is not a valid "class::method" string.', $controller)
+            );
+        }
+
+        $className = $match[1];
+        $controllerName = $match[2];
+        $actionName = $match[3];
```

Comment:
Could probably be `list(, $className, $controllerName, $actionName) = $match;`

Reply:
Started with `[, $className, $controllerName, $actionName] = $match` and this unfortunately didn't work. Yours do, thanks!